### PR TITLE
Improve in-code documentation of map primitives

### PR DESCRIPTION
**Pull request by ueco-jb**
closes https://github.com/CosmWasm/cw-plus/issues/407

It's not a world-breaking change, but can improve work a bit - when IDE shows suggestion about structures, it often shows Rust's doc comments as well.

### CODE REVIEW

File: packages/storage-plus/src/snapshot/mod.rs [hunk 1]
```diff
@@ -10,6 +10,9 @@ use cosmwasm_std::{Order, StdError, StdResult, Storage};
 use serde::de::DeserializeOwned;
 use serde::{Deserialize, Serialize};
 
+/// Structure holding a map of checkpoints composited from
+/// height as U64Key and checkpoint id.
```
Comment (ueco-jb) [inline packages/storage-plus/src/snapshot/mod.rs:14]: Is this correct? I have hard time understanding purpose of this `u32`.

Comment (ethanfrey) [reply packages/storage-plus/src/snapshot/mod.rs:14]: u32 is the data stored there.

It is a counter of how many times it has been checkpointed (maybe 2 users checkpointed, 1 released, but this stays checkpointed).

Good to add that as a comment

File: packages/storage-plus/src/snapshot/mod.rs [hunk 2]
```diff
@@ -23,6 +26,11 @@ pub(crate) struct Snapshot<'a, K, T> {
 }
 
 impl<'a, K, T> Snapshot<'a, K, T> {
+    /// Example:
```
Comment (ethanfrey) [inline packages/storage-plus/src/snapshot/mod.rs:29]: This is only used crate internal, so these docs are for other internal devs

**Review by ethanfrey — COMMENTED** [review]
Nice start. Some feedback

File: packages/storage-plus/src/snapshot/mod.rs (hunk 1 above)
Comment (ueco-jb) [reply packages/storage-plus/src/snapshot/mod.rs]: I've added that info as comment.

File: packages/storage-plus/src/snapshot/mod.rs (hunk 2 above)
Comment (ueco-jb) [reply packages/storage-plus/src/snapshot/mod.rs]: I removed this whole comment.

**Comment by ueco-jb** [comment]
I had plenty of issues while compiling those examples I've added, but finally I've managed to keep it small and precise.

**Review by ethanfrey — APPROVED** [review]
Looks good and brings knowledge from test code closer to the docs.

**Pull request merged**